src/node/channels.h

```diff
@@ -236,8 +239,8 @@ namespace ccf
         return false;
       }
 
-      auto ret =
-        recv_key->decrypt(header.get_iv(), header.tag, cipher, aad, plain.p);
+      auto ret = recv_key->decrypt(
```

Comment:
Passing a single pointer for plain.data() and having the size be implicit is dodgy, and having a single call do both is not great either. We should split it and make the size explicit.